### ISSUE TITLE
handle exceptions to joinspaces like Mr.

### ISSUE THREAD

**Issue by jaki** [vim/vim#6094]
**Is your feature request related something that is currently hard to do? Please describe.**

Yes, I think it is hard to do.  This may be an incompatible change that affects other plugins (see `runtime/pack/dist/opt/justify/plugin/justify.vim`).  It may be a strain on the join lines algorithm that is not worth it.

**Describe the solution you'd like**

When joining lines with `joinspaces` on, I do not want to see things like

```
Mr.  Jones
```

or

```
(e.g.  apples)
```

(note the double space).  It is easy to come across when formatting paragraphs (`gqap`) and hard to notice when it occurs.  There should be some list of words that are exempt from double spacing through `joinspaces`.  This can be an internal list (so that it can grow as we find more eligble words like `i.e.`) or an external list system (like spell files).

**Describe alternatives you've considered**

None.  Of course, one can turn off `joinspaces`, but I prefer it on.

**Additional context**

Here's a repro:

1. Start with the following text:

   ```
   My name is Mr.
   Jones and I like fruits (e.g.
   apples).
   ```

1. Set `textwidth` to 79
1. Run `gqap` and get:

   ```
   My name is Mr.  Jones and I like fruits (e.g.  apples).
   ```

**Comment by tejr** [comment]
Have you seen the `p` flag for `'formatoptions'` in `:help fo-table`?  It was added in [v8.1.0728](https://github.com/vim/vim/releases/tag/v8.1.0728).  It prevents breaking lines after a period if it's followed by only one space.  This is how GNU Emacs `auto-fill-mode` handles the problem.

From `:help fo-table`:

```
p       Don't break lines at single spaces that follow periods.  This is
        intended to complement 'joinspaces' and cpo-J, for prose with
        sentences separated by two spaces.  For example, with 'textwidth' set
        to 28:
                Surely you're joking, Mr. Feynman!
        Becomes:
                Surely you're joking,
                Mr. Feynman!
        Instead of:
                Surely you're joking, Mr.
                Feynman!
```

It doesn't directly treat your example case, but assuming Vim is the tool that wrapped the text narrowly in the first place, it might help avoid the situation coming up.

**Comment by jaki** [comment]
@tejr, yes, I have seen that flag.  In fact, that's where I got the `Mr.` example.  I don't think that option is addressing this issue though; my repro is still valid with `set formatoptions+=p`.

**Comment by schaiba** [comment]
Hello, can this be closed?
